src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/service/LimitedPartnershipService.java

```diff
@@ -22,14 +25,24 @@ public LimitedPartnershipService(LimitedPartnershipMapper mapper, LimitedPartner
         this.repository = repository;
     }
 
-    public String createLimitedPartnership(LimitedPartnershipSubmissionDto limitedPartnershipSubmissionDto, String requestId, String userId) {
+    public String createLimitedPartnership(Transaction transaction, LimitedPartnershipSubmissionDto limitedPartnershipSubmissionDto, String requestId, String userId) {
         ApiLogger.debug("Called createLimitedPartnership(...)");
 
         LimitedPartnershipSubmissionDao dao = mapper.dtoToDao(limitedPartnershipSubmissionDto);
         dao.setCreatedAt(LocalDateTime.now());
         dao.setUserId(userId);
+
         LimitedPartnershipSubmissionDao insertedSubmission = repository.insert(dao);
-        ApiLogger.infoContext(requestId, String.format("Limited Partnership Submission created with limited-partnership submission id: %s", insertedSubmission.getId()));
+
+        // Create the self-link
+        var selfLink = String.format("/transactions/%s/limited_partnership/partnership/%s", transaction.getId(), dao.getId());
```

Comment:
Sorry @HatiDev spotted this late on, but the URL on that sub-task isn't quite correct as we agreed 'limited_partnership' should be 'limited-partnership' (architect spec wasn't correct). Could you please change that?

Reply:
used the correct format to use - instead of _ within (limited-partnership)